File: kafka/src/main/java/com/example/kafka/service/KafkaConsumerService.java
package com.example.kafka.service;

import java.util.logging.Logger;
import org.springframework.kafka.annotation.KafkaListener;
import org.springframework.stereotype.Service;

@Service
public class KafkaConsumerService {

  private final Logger logger = Logger.getLogger(this.getClass().getName());

  @KafkaListener(topics = "my_topic", groupId = "my_group_id")
  public void consume(String message) {
  logger.info("Consumed message: " + message);
  }
}

File: kafka/src/main/kotlin/com/example/kafka/service/KafkaConsumerService.kt
package com.example.kafka.service

import com.example.kafka.dto.ProductEvent
import org.springframework.kafka.annotation.KafkaListener
import org.springframework.stereotype.Service
import org.slf4j.LoggerFactory.getLogger

@Service
class KafkaConsumerService {
    private val log = getLogger(this::class.java)

    @KafkaListener(topics = ["att-product-manage"], groupId = "product-group")
    fun consume(event: ProductEvent?) {
        log.info("이벤트 받아따~ : {}", event)
    }

}